### PR TITLE
 Added a new button to create a new circuit in insert Subcircuit dialog

### LINKED ISSUE #1751: Button to create "New Circuit" in "Insert Subcircuit" Dialogue
**Is your feature request related to a problem? Please describe.**
If there is only 1 circuit and we press ```Insert SubCircuit``` we get a dialogue box. Adding a button to ```Create a New Circuit``` can be helpful.

**Describe the solution you'd like**
A simply button or anchor tag to create a new circuit.

**Additional context**
Screenshot
image

**Are you working on this? (Yes/No)**
No.

### PR DESCRIPTION
**Pull request by Prerna-0202**
Fixes #1751 

#### Describe the changes you have made in this PR -
Added a new button `New Circuit +` to create a new circuit in the `Insert Subcircuit` dialogue box.

### Screenshots of the changes (If any) -
P2
pp

### CODE REVIEW

**Comment by Prerna-0202** [comment]
@shubhankarsharma00 , Sir Thank you for suggesting the changes .I have made the changes in my pr  . Now please review it.

**Comment by shubhankarsharma00** [comment]
looks like you tried to merge master and the feature branch. you can make a new PR. or try to fix history for this one. Look into git reset -soft and if you are able fix the history then force push the changes. It's upto you if you want to make a new PR though just mention this there and close this one.

**Comment by gr455** [comment]
As Shubhankar said, you can try and fix this branch by doing `git reset --soft HEAD~n` where n is the number of commits you want reset. This will remove these commits and add the changes in these commits to staging. You can then git reset <filename>to unstage individual changes.

If this is too inconvenient, you can also close this PR and create a new one

**Comment by Prerna-0202** [comment]
@gr455 @shubhankarsharma00, I have made the changes . Can u please now review my pr and provide me with the feedback.

**Review by gr455 — CHANGES_REQUESTED** [review]
Looks good just one small issue. Let's say I create `circuit2` and put `Main` as a subcircuit in `circuit2`. then trying to insert subcircuits in `Main` should also show `No subcircuits` (since the only other circuit `circuit2` has `Main` as a dependency). But right now the UI bugs out and shows `Main` when inserting subcircuit in `Main`

**Comment by Prerna-0202** [comment]
Okay @gr455,  I will commit the changes as soon as possible.

**Comment by Prerna-0202** [comment]
@gr455 I have commit the changes and now it's working fine. Can you please review my pr ...

**Review by gr455 — APPROVED** [review]
Looks good to me

**Comment by Prerna-0202** [comment]
@gr455 @tachyons, Do I need to make more changes in this pr??

File: app/views/simulator/edit.html.erb [hunk 1]
```diff
@@ -528,7 +529,8 @@
   id="insertSubcircuitDialog"
   class="subcircuitdialog"
   title="Insert SubCircuit"
-></div>
+  ></div>
+<div id="insertSubcircuitcontent"><span><p>Looks like there are no other circuits which doesn't have this circuit as a dependency. Create a new one!</p></span><a class="btn btn-xs custom-btn--primary dropdown-item text-center logixButton pl-1" id="newCircuit" onclick ='return $("#insertSubcircuitDialog").dialog("close");'>New Circuit +</a></div>
```
[the remark this answers is not in the document]
Comment (tachyons) [reply app/views/simulator/edit.html.erb:532]: - Make the text translatable using i18n 
- move inline onclick if feasible

**Comment by Prerna-0202** [comment]
@tachyons, I have commit the changes by making the text translatable using i18n...and the second changes you said I think it's not feasible or if you have any idea please tell...

**Comment by Prerna-0202** [comment]
@tachyons, I have commit the changes by making the text translatable using i18n...and the second changes you said I think it's not feasible or if you have any idea please tell...

**Comment by Prerna-0202** [comment]
@tachyons I have moved the inline onclick and resolved bot issues. Please review.

**Pull request merged**